### PR TITLE
[Fix] replace mmcv's function and modules imported with mmengine's

### PR DESCRIPTION
**Pull request by chhluo**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

1. use mmengine's `Config` `load_state_dict`, `FileClient`, `dump`, `load_checkpoint` and etc.
2. changes due to https://github.com/open-mmlab/mmcv/pull/2216, https://github.com/open-mmlab/mmcv/pull/2217, https://github.com/open-mmlab/mmengine/pull/447
## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: tests/test_datasets/test_cityscapes.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import os
 import unittest
 
-from mmcv import dump
+from mmengine import dump
```
Comment (RangiLyu) [inline tests/test_datasets/test_cityscapes.py:5]:
```suggestion
from mmengine.fileio import dump
```

File: tests/test_datasets/test_coco_api_wrapper.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import tempfile
 import unittest
 
-import mmcv
+from mmengine import dump
```
Comment (RangiLyu) [inline tests/test_datasets/test_coco_api_wrapper.py:5]:
```suggestion
from mmengine.fileio import dump
```

File: mmdet/models/backbones/swin.py [hunk 3]
```diff
@@ -9,10 +9,10 @@
 import torch.utils.checkpoint as cp
 from mmcv.cnn import build_norm_layer
 from mmcv.cnn.bricks.transformer import FFN, build_dropout
-from mmcv.runner import _load_checkpoint
 from mmengine.model import BaseModule, ModuleList
 from mmengine.model.utils import (constant_init, trunc_normal_,
                                   trunc_normal_init)
+from mmengine.runner.checkpoint import _load_checkpoint
```
Comment (RangiLyu) [inline mmdet/models/backbones/swin.py:15]: should not use the private function

**Comment by zhouzaida** [comment]
MMCV dev-2.x removes many functions in mmcv/utils and move them to mmengine. More detail can be found at https://github.com/open-mmlab/mmcv/pull/2217 and https://github.com/open-mmlab/mmengine/pull/447.

**Comment by hhaAndroid** [comment]
@chhluo Please synchronize the latest mmdet dev-3.x. There is a PR merged, it also needs to be modified.

**Comment by chhluo** [comment]
> @chhluo Please synchronize the latest mmdet dev-3.x. There is a PR merged, it also needs to be modified.

Done

**Review by hhaAndroid — APPROVED** [review]

**Review by ZwwWayne — APPROVED** [review]

**Comment by hoseinCode** [comment]
hi what is replace function for mmcv.runner?

**Pull request merged**